### ISSUE TITLE
Absence of Fontconfig breaks things besides just the HUD

### ISSUE THREAD

**Issue by nlburgin** [clementgallet/libTAS#210]
Fontconfig is supposedly an _optional_ dependency only needed for the HUD.

However, custom builds without it would fail to launch Celeste, with terminal outputting

```
Attempt 1: Couldn't connect to socket.
Attempt 2: Couldn't connect to socket.
Attempt 3: Couldn't connect to socket.
Attempt 4: Couldn't connect to socket.
Attempt 5: Couldn't connect to socket.
Attempt 6: Couldn't connect to socket.
Attempt 7: Couldn't connect to socket.
Attempt 8: Couldn't connect to socket.
Attempt 9: Couldn't connect to socket.
Attempt 10: Couldn't connect to socket.
The connection to the game was lost. Exiting
```

I initially assumed it was some sort of regression from using latest interim, so I reverted to 1.3.3 ( commit 5ca20a59239bc1852df5d410ca043ca6a5ea8283)

However, that did absolutely nothing. My custom build was failing to function correctly, even when it should have been identical the 1.3.3 binary release (which did work).

The output of cmake when generating the build system did not indicate any libraries missing except fontconfig.

I then tried installing `libfontconfig1-dev` and regenerating the build directory, which completely fixed the problem. 

It's looking like this "optional" dependency is actually mandatory.

I see two possibilities here:

**1)** There's some obscure preprocessor bug somewhere that causes a bug to manifest only when an ifdef block isn't triggered. Sort of similar in nature to the one I fixed in #205, except it causes weird freakouts at runtime instead of just an obvious compile error.

**2)** This is actually somehow a hard dependency, in which case the readme shouldn't make it look more optional than the rest of them by saying it's only for the HUD.

**Comment by clementgallet** [comment]
There was some problem with GL functions defined inside the `#ifdef LIBTAS_ENABLE_HUD` macro and used elsewhere, this is fixed in 8a50522926834e747a58519bdc7755633ee2310d . Could you test it again?

**Comment by nlburgin** [comment]
Yes, that fixes it.

Though due to #212 I'm not using the actual latest master, but the results of 

```
$ git branch working 97324836b9b86cc79a9ff16b90acf2d34acf281b
$ git checkout working
$ git cherry-pick 8a50522926834e747a58519bdc7755633ee2310d
```
as seen [here](https://github.com/nlburgin/libTAS/commits/working).
